### ISSUE TITLE
Some samples tests are consistently failing

### ISSUE THREAD

**Issue by skuruppu** [googleapis/nodejs-spanner#1263]
I noticed that the samples system test presubmit is failing consistently with the following error:

```
  1) Spanner
       "before all" hook for "should create an example database":
     TypeError: Cannot read property 'filter' of undefined
      at /workspace/samples/system-test/spanner.test.js:102:12
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
      at async Promise.all (index 1)
      at async deleteStaleInstances (system-test/spanner.test.js:88:3)
      at async Context.<anonymous> (system-test/spanner.test.js:125:5)

  2) Spanner
       "after all" hook for "should insert with custom timeout and retry settings":
     Error: 5 NOT_FOUND: Instance not found: projects/long-door-651/instances/test-instance-1604571392
      at Object.callErrorFromStatus (/workspace/node_modules/@grpc/grpc-js/build/src/call.js:31:26)
          -> /workspace/node_modules/@grpc/grpc-js/src/call.ts:81:24
      at Object.onReceiveStatus (/workspace/node_modules/@grpc/grpc-js/build/src/client.js:176:52)
          -> /workspace/node_modules/@grpc/grpc-js/src/client.ts:334:36
      at /workspace/node_modules/@grpc/grpc-js/build/src/call-stream.js:65:35
          -> /workspace/node_modules/@grpc/grpc-js/src/call-stream.ts:138:27
      at Object.onReceiveStatus (/workspace/node_modules/grpc-gcp/build/src/index.js:73:29)
          -> /workspace/node_modules/grpc-gcp/src/index.ts:129:15
      at InterceptingListenerImpl.onReceiveStatus (/workspace/node_modules/@grpc/grpc-js/build/src/call-stream.js:60:23)
          -> /workspace/node_modules/@grpc/grpc-js/src/call-stream.ts:134:19
      at Object.onReceiveStatus (/workspace/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:342:141)
          -> /workspace/node_modules/@grpc/grpc-js/src/client-interceptors.ts:434:34
      at Object.onReceiveStatus (/workspace/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:305:181)
          -> /workspace/node_modules/@grpc/grpc-js/src/client-interceptors.ts:397:48
      at /workspace/node_modules/@grpc/grpc-js/build/src/call-stream.js:124:78
          -> /workspace/node_modules/@grpc/grpc-js/src/call-stream.ts:237:24
      at processTicksAndRejections (internal/process/task_queues.js:79:11)
```

**Comment by skuruppu** [comment]
@AVaksman would you be able to please take a look?

**Comment by AVaksman** [comment]
I believe this issue is resolved with #1267 
#1262 seems to be all green now

**Comment by skuruppu** [comment]
Thanks for looking into this @AVaksman.
